src/cose.c

```diff
@@ -514,11 +535,118 @@ static DPS_Status EncodeKDFContext(DPS_TxBuffer* buf, int8_t alg, uint8_t keyLen
+    case DPS_KEY_EC:
+        if (ckey->type != COSE_KEY_EC) {
+            DPS_ERRPRINT("Provided key has invalid type %d\n", key->type);
+            return DPS_ERR_MISSING;
+        }
+        switch (key->ec.curve) {
+        case DPS_EC_CURVE_P256: len = 32; break;
+        case DPS_EC_CURVE_P384: len = 48; break;
+        case DPS_EC_CURVE_P521: len = 66; break;
+        default:
+            DPS_ERRPRINT("Provided key has unsupported curve %d\n", key->ec.curve);
+            return DPS_ERR_MISSING;
+        }
+        memset(&ckey->ec, 0, sizeof(ckey->ec));
+        ckey->ec.curve = key->ec.curve;
+        if (key->ec.x) {
+            memcpy_s(ckey->ec.x, sizeof(ckey->ec.x), key->ec.x, len);
+        }
+        if (key->ec.y) {
+            memcpy_s(ckey->ec.y, sizeof(ckey->ec.y), key->ec.y, len);
+        }
+        if (key->ec.d) {
+            memcpy_s(ckey->ec.d, sizeof(ckey->ec.d), key->ec.d, len);
+        }
+        break;
+    case DPS_KEY_EC_CERT:
+        if (ckey->type != COSE_KEY_EC) {
+            DPS_ERRPRINT("Provided key has invalid type %d\n", key->type);
+            return DPS_ERR_MISSING;
+        }
+        if (key->cert.privateKey) {
+            return ParsePrivateKey_ECDSA(key->cert.privateKey, key->cert.privateKeyLen,
+                                         key->cert.password, key->cert.passwordLen,
+                                         &ckey->ec.curve, ckey->ec.d);
+        } else {
+            return ParseCertificate_ECDSA(key->cert.cert, key->cert.certLen,
+                                          &ckey->ec.curve, ckey->ec.x, ckey->ec.y);
+        }
+        break;
+    default:
+        DPS_ERRPRINT("Unsupported key type %d\n", key->type);
+        return DPS_ERR_MISSING;
+    }
+    return DPS_OK;
+}
+
+static DPS_Status GetKey(DPS_KeyStore* keyStore, const unsigned char* kid, size_t kidLen, COSE_Key* key)
+{
+    DPS_KeyStoreRequest request;
+
+    if (!keyStore || !keyStore->keyHandler) {
+        return DPS_ERR_MISSING;
+    }
+    memset(&request, 0, sizeof(request));
+    request.keyStore = keyStore;
+    request.data = key;
+    request.setKey = SetKey;
+    return keyStore->keyHandler(&request, kid, kidLen);
+}
+
+static DPS_Status GetEphemeralKey(DPS_KeyStore* keyStore, COSE_Key* key)
+{
+    DPS_KeyStoreRequest request;
+    DPS_Key k;
+
+    if (!keyStore || !keyStore->ephemeralKeyHandler) {
+        return DPS_ERR_MISSING;
+    }
+    memset(&request, 0, sizeof(request));
+    request.keyStore = keyStore;
+    request.data = key;
+    request.setKey = SetKey;
+    switch (key->type) {
+    case COSE_KEY_SYMMETRIC:
+        k.type = DPS_KEY_SYMMETRIC;
+        break;
+    case COSE_KEY_EC:
+        k.type = DPS_KEY_EC;
+        k.ec.curve = key->ec.curve;
+        break;
+    default:
+        return DPS_ERR_MISSING;
+    }
+    return keyStore->ephemeralKeyHandler(&request, &k);
+}
+
+static DPS_Status GetSignatureKey(DPS_KeyStore* keyStore, const Signature* sig, COSE_Key* key)
```

Comment:
Should this be called GetSigningKey?

Reply:
I wanted the name to match that it's getting the key identified by the Signature struct.